store/postgres/src/chain_store.rs

```diff
@@ -1028,6 +1039,68 @@ mod data {
                 .collect())
         }
 
+        pub(super) fn clear_call_cache(
+            &self,
+            conn: &PgConnection,
+            from: Option<i32>,
+            to: Option<i32>,
+        ) -> Result<(), Error> {
+            if from.is_none() && to.is_none() {
+                // fast path to clear the call cache table.
+                self.truncate_call_cache(conn)?;
+                return Ok(());
+            }
+            match self {
+                Storage::Shared => {
+                    use public::eth_call_cache as cache;
+                    let mut delete_stmt = diesel::delete(cache::table).into_boxed();
+                    if let Some(from) = from {
+                        delete_stmt = delete_stmt.filter(cache::block_number.ge(from));
+                    }
+                    if let Some(to) = to {
+                        delete_stmt = delete_stmt.filter(cache::block_number.le(to))
+                    }
+                    delete_stmt.execute(conn).map_err(Error::from)?;
+                    Ok(())
+                }
+                Storage::Private(Schema { call_cache, .. }) => match (from, to) {
```

Comment:
```suggestion
                // Because they are dynamically defined, our private call cache tables can't
                // implement all the required traits for deletion. This means we can't use Diesel
                // DSL with them and must rely on the `sql_query` function instead.
                Storage::Private(Schema { call_cache, .. }) => match (from, to) {
```